pkg/registry/apis/alerting/notifications/receiver/authorize.go

```diff
@@ -20,19 +20,24 @@ func Authorize(ctx context.Context, ac accesscontrol.AccessControl, attr authori
 
 	var action accesscontrol.Evaluator
 	switch attr.GetVerb() {
+	case "create":
+		action = accesscontrol.EvalAny(
+			accesscontrol.EvalPermission(accesscontrol.ActionAlertingNotificationsWrite),
+			accesscontrol.EvalPermission(accesscontrol.ActionAlertingReceiversCreate),
+		)
 	case "patch":
 		fallthrough
-	case "create":
-		fallthrough // TODO: Add alert.notifications.receivers:create permission
 	case "update":
 		action = accesscontrol.EvalAny(
-			accesscontrol.EvalPermission(accesscontrol.ActionAlertingNotificationsWrite), // TODO: Add alert.notifications.receivers:write permission
+			accesscontrol.EvalPermission(accesscontrol.ActionAlertingNotificationsWrite),
+			accesscontrol.EvalPermission(accesscontrol.ActionAlertingReceiversUpdate),
 		)
 	case "deletecollection":
 		fallthrough
 	case "delete":
 		action = accesscontrol.EvalAny(
-			accesscontrol.EvalPermission(accesscontrol.ActionAlertingNotificationsWrite), // TODO: Add alert.notifications.receivers:delete permission
+			accesscontrol.EvalPermission(accesscontrol.ActionAlertingNotificationsWrite),
+			accesscontrol.EvalPermission(accesscontrol.ActionAlertingReceiversDelete),
```

Comment:
It would be simple to add the `ReceiverAccess` service here, what's not clear to me is what kind of responsibility it should have. For example:
- It is responsible for completely determining auth:
  - This is easy enough when the request is for a single resource with a UID, but what about those without a clear name. For example: does `create` have a value in`attr.GetName()`, `list` response filtering would need to happen after the list is retrieved.
- It is responsible for just preconditon checks:
  - This should be straightforward but the benefits aren't clear to me as it's redundant with the legacy storage and not sufficient for the unified storage.

Reply:
I think the authz process should be multi-tier, sooner we reject an unauthorized request, better, the authorization logic in domain service also makes sense to prevent unauthorized access via dependencies.
I think create action does not need UID scope because it does not seem to make sense because it's awkward to be able to pre-authorize creation of a receiver with a specific name. List operation can still use `readRedactedReceiversPreConditionsEval`